V9 Configuration documentation (#3278)
umbraco/UmbracoDocs: Reference/V9-Config/index-v9.md

---
versionFrom: 9.0.0
meta.Title: "Umbraco configuration"
meta.Description: "Information on configuring Umbraco"
state: complete
verified-against: beta-3
update-links: true
---

# Configuration Files

> In V9, we have moved away from the previous configuration using .config files, now instead using the netcore built in configuration pattern. This means that there is no longer seperate files for different configuration, the configuration is now primarily done from the `appsettings.json` file.

Reviewer: ```suggestion
In V9, we have moved away from the previous configuration using .config files, to instead using the netcore built-in configuration pattern. This means that there is no longer separate files for different configuration, the configuration is now primarily done from the `appsettings.json` file.
```

For more in depth information on the configuration pattern see Microsofts [Configuration in ASP.NET Core](https://docs.microsoft.com/en-us/aspnet/core/fundamentals/configuration/?view=aspnetcore-5.0) article.

## Managing configuration

> You might not always want to have the configuration stored in the `appsettings.json` file, for instance you might not want to have the admin password in the file if using the unattended feature. You might also want to use a specific set of configurations when developing your solution, fortunately the `IConfiguration` pattern.

Reviewer: ```suggestion
You might not always want to have the configuration stored in the `appsettings.json` file, for instance, you might not want to have the admin password in the file if using the unattended feature. You might also want to use a specific set of configurations when developing your solution, fortunately, the `IConfiguration` pattern.
```

With the configuration pattern the settings can be read from multiple different source, where some take precedence over other, you can configure you site with: 

1. The `appsettings.json` file
2. An `appsettings.{environment}.json` file
3. UserSecrets (Only when in development)
4. Environment variables
5. Command line arguments

This list is in order of precedence, so the values from `appSettings.json` will only be used if they're not also define in the environment variables, if they are then the environment variable will be used instead.

> There isone caveat, to this precedence though, the `appSettings.{environment}.json` file will only be used if the current environment matches the name of the config file, for instance the `appsettings.Development.json` file will only be used when the environment is set to development.

Reviewer: ```suggestion
There is one caveat, to this precedence though, the `appSettings.{environment}.json` file will only be used if the current environment matches the name of the config file, for instance, the `appsettings.Development.json` file will only be used when the environment is set to development.
```
